### PR TITLE
Fixed null token panic from 'v1/auth/token/' endpoints and returned p…

### PR DESCRIPTION
**Pull request by ryowright**
Fixes null token panic when hitting **/v1/auth/token/lookup, /v1/auth/token/renew, /v1/auth/token/revoke, /v1/auth/token/revoke-orphan** endpoints and returns proper error response [(Issue #13225)](https://github.com/hashicorp/vault/issues/13225).

### CODE REVIEW

**Comment by hsimon-hashicorp** [comment]
And thanks for this one too! Similar story here, I'll get a review going soon.

File: vault/request_handling.go [hunk 1]
```diff
@@ -543,6 +543,9 @@ func (c *Core) handleCancelableRequest(ctx context.Context, req *logical.Request
 				}
 				break
 			}
+			if token == nil {
+				return logical.ErrorResponse("bad token"), logical.ErrPermissionDenied
```
Comment (pmmukh) [inline vault/request_handling.go:547]: maybe `invalid token` might be better? cause browsing through current error responses, invalid seems to be the word mostly used for erroneous input.

**Review by pmmukh — APPROVED** [review]
one nit, lgtm! And thanks for submitting this PR!

File: vault/request_handling.go (hunk 1 above)
Comment (ryowright) [reply vault/request_handling.go]: Hi, I just committed a change to the error message, changing it to invalid token. I accidentally wrote this as 'null token' in the commit message, but in the code it is 'invalid token' as requested.

**Comment by pmmukh** [comment]
Thanks for submitting that change, looks good! Could you please rebase this PR off main once ? There seems to be some conflicts, and a couple failing tests that I believe should be fixed on main now.

**Comment by pmmukh** [comment]
@ryowright I see you commited some changes in the vault-data folder, which I assume is a file backend you were using for testing, could you please revert those changes ?

**Comment by ryowright** [comment]
I apologize, I missed removing the vault-data folder, which I usually do. Will revert it right now.

**Comment by pmmukh** [comment]
Thanks for removing that folder, the pr looks good to me now! Merging it in, and thanks for the submission!

**Pull request merged**